TrkReco/src/TrkRecoMcUtils_tool.cc

```diff
@@ -29,229 +29,214 @@
-//    ++n_gen_hits;
-//    double pstep = step->momentum().mag();
-//    if (pstep > pEntrance) {
-//      pEntrance = pstep;
-//    }
-//  }
-//       }
-//     }
+  //  sim_id        = simptr->id().asInt();
+  //       }
 
-//     if (pEntrance < 100. ) n_gen_hits = 0;
+  //       if ((gen_index > 0) && (sim_id == 1)) {
+  //  double step_time = timeOffsets->timeWithOffsetsApplied(*step);
+  //  step_time = fmod(step_time,_mbtime);
+  //  if (step_time > time_threshold) {
+  //    ++n_gen_hits;
+  //    double pstep = step->momentum().mag();
+  //    if (pstep > pEntrance) {
+  //      pEntrance = pstep;
+  //    }
+  //  }
+  //       }
+  //     }
 
-//     return n_gen_hits;
-//   }
+  //     if (pEntrance < 100. ) n_gen_hits = 0;
 
-//-----------------------------------------------------------------------------
-// const StrawDigiMCCollection* TrkRecoMcUtils::getListOfMcStrawHits(const art::Event* Event,const
-// art::InputTag& Tag) {
-//   auto handle = Event->getValidHandle<StrawDigiMCCollection>(Tag);
-//   const StrawDigiMCCollection* coll = handle.product();
-//   return coll;
-// }
+  //     return n_gen_hits;
+  //   }
 
-//-----------------------------------------------------------------------------
-const SimParticle* TrkRecoMcUtils::getSimParticle(const art::Event* Event, int HitIndex) {
+  //-----------------------------------------------------------------------------
+  // const StrawDigiMCCollection* TrkRecoMcUtils::getListOfMcStrawHits(const art::Event* Event,const art::InputTag& Tag) {
+  //   auto handle = Event->getValidHandle<StrawDigiMCCollection>(Tag);
+  //   const StrawDigiMCCollection* coll = handle.product();
+  //   return coll;
+  // }
 
-  if (Event->event() != _lastEvent)
-    initEvent(Event);
+  //-----------------------------------------------------------------------------
+  const SimParticle* TrkRecoMcUtils::getSimParticle(const art::Event* Event, int HitIndex) {
 
-  return (*_mcdigis)[HitIndex].earlyStrawGasStep()->simParticle().get();
-}
+    if (Event->event() != _lastEvent) initEvent(Event);
 
-//-----------------------------------------------------------------------------
-int TrkRecoMcUtils::getID(const SimParticle* Sim) { return Sim->id().asInt(); }
+    return (*_mcdigis)[HitIndex].earlyStrawGasStep()->simParticle().get();
+  }
 
-//-----------------------------------------------------------------------------
-int TrkRecoMcUtils::getPdgID(const SimParticle* Sim) { return Sim->pdgId(); }
+  //-----------------------------------------------------------------------------
+  const XYZVectorF* TrkRecoMcUtils::getMom(const art::Event* Event, int HitIndex) {
 
-//-----------------------------------------------------------------------------
-float TrkRecoMcUtils::getStartMom(const SimParticle* Sim) {
-  CLHEP::HepLorentzVector const& p = Sim->startMomentum();
-  return sqrt(p.x() * p.x() + p.y() * p.y() + p.z() * p.z());
-}
+    if (Event->event() != _lastEvent) initEvent(Event);
```

Comment:
This test is not robust.  You should compare Event->id();  .  In a sparse skim in which there are sequential events from different run/subrun but with the same event number, you will not detect that you have a new event and will fail to call initEvent.  I see that this is ubiquitous in this file.  Please fix this - it's straightforward.

Reply:
done